feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/clustering/web/main/module.xml

```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2018, Red Hat, Inc., and individual contributors
+  ~ as indicated by the @author tags. See the copyright.txt file in the
+  ~ distribution for a full listing of individual contributors.
+  ~
+  ~ This is free software; you can redistribute it and/or modify it
+  ~ under the terms of the GNU Lesser General Public License as
+  ~ published by the Free Software Foundation; either version 2.1 of
+  ~ the License, or (at your option) any later version.
+  ~
+  ~ This software is distributed in the hope that it will be useful,
+  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  ~ Lesser General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Lesser General Public
+  ~ License along with this software; if not, write to the Free
+  ~ Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  -->
+
+<module xmlns="urn:jboss:module:1.8" name="org.wildfly.extension.clustering.web">
+    <properties>
+        <property name="jboss.api" value="private"/>
+    </properties>
+
+    <resources>
+        <artifact name="${org.wildfly:wildfly-clustering-web-extension}"/>
+    </resources>
+
+    <dependencies>
+        <module name="javax.api"/>
+        <module name="javax.servlet.api"/>
+
+        <module name="org.infinispan"/>
+        <module name="org.infinispan.commons"/>
+        <module name="org.jboss.as.clustering.common"/>
+        <module name="org.jboss.as.controller"/>
+        <module name="org.jboss.as.ee"/>
+        <module name="org.jboss.as.server"/>
+        <module name="org.jboss.as.web-common"/>
+        <module name="org.jboss.dmr"/>
+        <module name="org.jboss.logging"/>
+        <module name="org.jboss.metadata.common"/>
+        <module name="org.jboss.metadata.web"/>
+        <module name="org.jboss.modules"/>
+        <module name="org.jboss.msc"/>
+        <module name="org.jboss.staxmapper"/>
+        <module name="org.jboss.threads"/>
+        <module name="org.jboss.vfs"/>
+        <module name="org.wildfly.clustering.ee.infinispan"/>
+        <module name="org.wildfly.clustering.ee.spi"/>
+        <module name="org.wildfly.clustering.infinispan.spi"/>
+        <module name="org.wildfly.clustering.service"/>
+        <module name="org.wildfly.clustering.web.container"/>
+        <module name="org.wildfly.clustering.web.infinispan"/>
+        <module name="org.wildfly.clustering.web.spi"/>
+        <module name="org.wildfly.clustering.web.undertow" services="import"/>
+        <module name="org.wildfly.common"/>
+        <module name="org.wildfly.security.elytron-private"/>
+    </dependencies>
```

Comment:
The following modules are not used to cover a direct dependency of this module:
- org.wildfly.security.elytron-private
- org.wildfly.clustering.ee.infinispan
- javax.servlet.api
- org.jboss.threads
- org.wildfly.common
- org.wildfly.clustering.ee.spi
- org.wildfly.clustering.web.undertow
- org.jboss.modules
- org.infinispan.commons
- org.wildfly.clustering.web.container

I checked only direct dependencies, I have not checked if they are used to supply a service loaded by this module or if they are required due to an external activation. That requires a wide knowledge of this subsystem, but maybe the list of required dependencies can be reduced using the above list.

Reply:
Thanks for the analysis - some of these are certainly required, but I'll verify the others.